src/lib/utils.ts

```diff
@@ -21,3 +21,14 @@ export const appUrl = readable<string>(PUBLIC_APP_URL, (set) => {
     }
   });
 });
+
+export const filterAttributes = (attributes: string[], input: object | object[]) => {
+  const filteredObject = (obj: object) => {
+    return Object.fromEntries(Object.entries(obj).filter(([key]) => attributes.includes(key)));
+  };
+  if (Array.isArray(input)) {
+    return input.map(filteredObject);
+  } else {
+    return filteredObject(input);
+  }
```

Comment:
Not a real significant performance concern but since it is a one-liner anyway we may as well skip declaring the `filteredObject` function
```suggestion
  if (Array.isArray(input)) {
    return input.map(filteredObject);
  } else {
    return Object.fromEntries(Object.entries(input).filter(([key]) => attributes.includes(key)));;
  }
```